src/CommunityToolkit.Maui.Core/Platform/KeyboardManager/KeyboardExtensions.shared.cs

```diff
@@ -0,0 +1,126 @@
+using System.Diagnostics.CodeAnalysis;
+using CommunityToolkit.Maui.Core.Extensions;
+using Microsoft.Maui.Dispatching;
+
+#if IOS || MACCATALYST
+using PlatformView = UIKit.UIView;
+#elif ANDROID
+using PlatformView = Android.Views.View;
+#elif WINDOWS
+using PlatformView = Microsoft.UI.Xaml.FrameworkElement;
+#elif TIZEN
+using PlatformView = Tizen.NUI.BaseComponents.View;
+#elif (NETSTANDARD || !PLATFORM) || (NET6_0_OR_GREATER && !IOS && !ANDROID && !TIZEN)
+using PlatformView = System.Object;
+using IPlatformViewHandler = Microsoft.Maui.IViewHandler;
+#endif
+
+namespace CommunityToolkit.Maui.Core.Platform;
+
+/// <summary>
+/// Extension methods for interacting with a platforms Soft Keyboard Pane
+/// </summary>
+public static partial class KeyboardExtensions
+{
+	static bool TryGetPlatformView(
+		this ITextInput textInput,
+		[NotNullWhen(true)] out PlatformView? platformView,
+		[NotNullWhen(true)] out IPlatformViewHandler? handler,
+		[NotNullWhen(true)] out IView? view)
+	{
+		if (textInput is not IView iView ||
+			iView.Handler is not IPlatformViewHandler platformViewHandler)
+		{
+			platformView = null;
+			handler = null;
+			view = null;
+			return false;
+		}
+
+		if (iView.Handler?.PlatformView is not PlatformView platform)
+		{
+			platformView = null;
+			handler = null;
+			view = null;
+			return false;
+		}
+
+		handler = platformViewHandler;
+		platformView = platform;
+		view = iView;
+
+		return true;
+	}
+
+	/// <summary>
+	/// If a soft input device is currently showing, this will attempt to hide it.
+	/// </summary>
+	/// <param name="targetView"></param>
+	/// <returns>
+	/// Returns <c>true</c> if the platform was able to hide the soft input device.</returns>
+	public static Task<bool> HideKeyboardAsync(this ITextInput targetView)
+	{
+		if (!targetView.TryGetPlatformView(
+			out var platformView,
+			out _,
+			out _))
+		{
+			return Task.FromResult(false);
```

Comment:
Here and in other places I can see a lot of hot paths, so what do you think to change the return type to `ValueTask`?

Reply:
I haven't used `ValueTask` much.

Thoughts from this post?
https://stackoverflow.com/a/43003779

> As such, the default choice for any asynchronous method should be to return a Task or Task<TResult>. Only if performance analysis proves it worthwhile should a ValueTask<TResult> be used instead of Task<TResult>.

Reply:
Here's an easy rule of thumb for `ValueTask`:
If the hot-path of your method not require/use the `await` keyword, you should return `ValueTask` / `ValueTask<T>` instead of `Task` / `Task<T>`

Reply:
What if a few months from now we do require/use the `await` keyboard? do you just break the API? or do you still just return it via `ValueTask` with the result?

My rule of thumb is if someone with the first name "Stephen" says to do something I just do that thing. 

That being said, I'm fine here with ValueTask. I don't have a strong argument for one vs the other.

Reply:
It is not a breaking change to move from `ValueTask` to `Task`.

It is a breaking change to move from `Task` to `ValueTask`, albeit it would break very few (if any) users. The biggest difference in using `ValueTask` is that you cannot re-use it and pass it around like you can with a `Task`; most users don't do advanced things like that with `Task` though.

Reply:
Well, we don't need to break it. So, the `ValueTask` can be used in async scenarios. The thing is, if the code resolves synchronously, we don't allocate anything to the work. Now if we need to complete it asynchronously then we will allocate a `Task` and that will work as normal `Task`. Also, if the user needs/want a `Task` behavior here, the dev can use `.AsTask()` extension method.